src/stores/index.ts

```diff
@@ -0,0 +1,158 @@
+import { ref, computed } from 'vue'
+import { defineStore } from 'pinia'
+import { ElMessage } from 'element-plus'
+import { io } from 'socket.io-client'
+import { i18n } from '@/locales'
+
+interface User {
+  id: string
+  name: string
+  card: string | null
+}
+
+interface Room {
+  id: string
+  name: string
+  users: { [key: string]: User }
+  cardVisible: boolean
+}
+
+const socket = io()
+
+const firstConnection = ref(true)
+
+const userName = ref(localStorage.getItem('userName'))
+const userSaved = ref(localStorage.getItem('userName') !== null)
+
+socket.on('connect', () => {
+  console.log(`connected to websocket with id ${socket.id}`)
+  if(!firstConnection.value) {
+    ElMessage(i18n.t('notifications.websocket-reconnected'))
+  }
+  if (userName.value) {
+    socket.emit('setUserName', userName.value)
+  } else {
+    socket.emit('whoAmI', (data: { userName: string }) => {
+      userName.value = data.userName
+    })
+  }
+})
+
+socket.on('disconnect', (reason) => {
+  firstConnection.value = false
+  ElMessage(i18n.t('notifications.websocket-disconnected'))
+  if (reason === "io server disconnect") {
+    // the disconnection was initiated by the server, you need to reconnect manually
+    setTimeout(function () {
+      socket.connect();
+    }, 3000)
+  }
+})
+
+export const useRoomStore = defineStore('store', () => {
+  const room = ref({} as Room)
+
+  function setUser(name: string) {
+    userName.value = name
+    userSaved.value = true
+    socket.emit('setUserName', name)
+    localStorage.setItem('userName', name)
+  }
+
+  const users = computed(() =>
+    !room.value.users
+      ? []
+      : Object.values(room.value.users).map((user: User) => {
+          if (user.id === socket.id) {
+            return { name: user.name, card: user.card }
+          } else {
+            if (room.value.cardVisible) {
+              return { name: user.name, card: user.card }
+            } else {
+              return { name: user.name, card: user.card ? '?' : null }
+            }
+          }
```

Comment:
else peut être supprimé (car return dans le if précédent):
```suggestion
          if (room.value.cardVisible) {
            return { name: user.name, card: user.card }
          }
          return { name: user.name, card: user.card ? '?' : null }
```